.github/files/assemble-image-tags.js

```diff
@@ -32,8 +32,11 @@ async function main(rockMetas){
                     partial + v.metadata.container.tags.filter(t => t.startsWith(rockVersion_)).length, 0
                 )
                 core.info(`Number of containers tagged ${owner}/${meta.name}/${rockVersion_}: ${patchRev}`)
-                core.info(`Tagging image ${meta.image} with ${meta.version}`)
                 meta.version = rockVersion_ + patchRev
+                if (versions.flatMap(v => v.metadata.container.tags).includes(meta.version)) {
```

Comment:
[`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) returns a new array with all the tags, duplicating the array in memory. In this case, we don't need the array after it's created, so we can use `some` which is part of the Array API combined with a sprinkle of `?.` operator to safely access those properties:
```suggestion
                if (versions.some(v => v.metadata?.container?.tags?.includes(meta.version)) {
```